ci_framework/roles/cifmw_cephadm/tasks/dashboard/dashboard.yml

```diff
@@ -0,0 +1,109 @@
+---
+# Copyright 2023 Red Hat, Inc.
+# All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+- name: Get ceph_cli
+  ansible.builtin.include_tasks: ceph_cli.yml
+  vars:
+    mount_spec: true
+    sensitive_data: true
+    mount_certs: "{{ (cifmw_cephadm_dashboard_protocol == 'https' and cifmw_cephadm_dashboard_crt | length > 0 and cifmw_cephadm_dashboard_key | length > 0) | bool }}"
+
+- name: Configure the Ceph Dashboard port
+  become: true
+  block:
+    - name: "set the dashboard port ({{ cifmw_cephadm_dashboard_port }})"
+      command: "{{ cifmw_cephadm_ceph_cli }} config set mgr mgr/dashboard/server_port {{ cifmw_cephadm_dashboard_port }}"
+      changed_when: false
+    - name: "set the dashboard SSL port ({{ cifmw_cephadm_dashboard_port }})"
+      command: "{{ cifmw_cephadm_ceph_cli }} config set mgr mgr/dashboard/ssl_server_port {{ cifmw_cephadm_dashboard_port }}"
+      run_once: true
+
+- name: disable SSL for dashboard
+  become: true
+  command: "{{ cifmw_cephadm_ceph_cli }} config set mgr mgr/dashboard/ssl false"
+  when: cifmw_cephadm_dashboard_protocol == "http"
+
+- name: Configure SSL key/cert for the Ceph Dashboard if provided
+  become: true
+  block:
+    - name: enable SSL for dashboard
+      command: "{{ cifmw_cephadm_ceph_cli }} config set mgr mgr/dashboard/ssl true"
+      run_once: true
+
+    - name: import dashboard certificate file
+      command: "{{ cifmw_cephadm_ceph_cli }} config-key set mgr/dashboard/crt -i {{ cifmw_cephadm_dashboard_crt }}"
```

Comment:
in case of tripleo with tls, grafana crt is set by defualt  but dashboard crt is set only if dashboard is enabled.
Now if dashboard is enabled by default, we can set both at once in the same block 
but do you think we need to converge the two different config keys below ?
'mgr/cephadm/grafana_crt'
'mgr/dashboard/crt'

I will address in the next iteration after the clarification

Reply:
This is mostly related to my previous comment. We always need to set the two different `config-keys` you mentioned, but to make the code simpler and easy to maintain, I think the best option we have is to point to the same crt/key variable.

Example:
```
cifmw_cephadm_dashboard_cert: <path_to crt>
cifmw_cephadm_dashboard_key: <path_to_key>
```

Then, when the path is provided, we can set the config-keys via ansible for both the dashboard and the monitoring stack, but reusing the same variables.
wdyt